localstack-core/localstack/testing/testselection/github.py

```diff
@@ -1,24 +1,41 @@
 import requests
 
+GITHUB_V3_JSON = "application/vnd.github.v3+json"
 
-def get_pr_details(repo_name: str, pr_number: str, token: str) -> (str, str):
+# We assume that test selection scripts are only run on repositories under the localstack organization
+REPO_OWNER = "localstack"
+
+
+def get_pr_details_from_number(repo_name: str, pr_number: str, token: str) -> (str, str):
     """
     Fetch the base commit SHA, and the head commit SHA of a given pull request number from a GitHub repository.
     """
-    url = f"https://api.github.com/repos/{repo_name}/pulls/{pr_number}"
-    headers = {"Accept": "application/vnd.github.v3+json"}
+    url = f"https://api.github.com/repos/{REPO_OWNER}/{repo_name}/pulls/{pr_number}"
+    headers = {"Accept": GITHUB_V3_JSON}
     if token:
         headers["Authorization"] = f"token {token}"
     pr_data = requests.get(url, headers=headers).json()
     return pr_data["base"]["sha"], pr_data["head"]["sha"]
 
 
-def get_pr_details_from_url(pr_url: str, token: str) -> (str, str):
+def get_pr_details_from_branch(repo_name: str, branch: str, token: str) -> (str, str):
+    """
+    Fetch the base commit SHA, and the head commit SHA of a given branch from a GitHub repository.
+    """
+    url = f"https://api.github.com/repos/{REPO_OWNER}/{repo_name}/pulls?head={REPO_OWNER}:{branch}"
+    headers = {"Accept": GITHUB_V3_JSON}
+    if token:
+        headers["Authorization"] = f"token {token}"
+    pr_data = requests.get(url, headers=headers).json()
+    print(f"Detected PR Number #{pr_data[0]['number']}")
+    return pr_data[0]["base"]["sha"], pr_data[0]["head"]["sha"]
```

Comment:
This section here does not handle any potential error cases at all. Most importantly, if there is just is no PR for the branch this will end up in an `IndexError`.
This this is a case which can happen frequently, it would be great if you could make this more explicit and handle these cases.

Reply:
Now throwing `ValueErrors` with more clear indications of what is wrong